src/components/AmountsForm.tsx

```diff
@@ -0,0 +1,109 @@
+import Grid from '@mui/material/Grid';
+import TextField from '@mui/material/TextField';
+import AddIcon from '@mui/icons-material/Add';
+import DeleteIcon from '@mui/icons-material/Delete';
+import IconButton from '@mui/material/IconButton';
+import type { IInvoiceForm } from './InvoiceForm.interface';
+import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
+import { type UseFormRegister, type Control, useFieldArray } from 'react-hook-form';
+
+type AmountsProps = {
+  register: UseFormRegister<IInvoiceForm>;
+  control: Control<IInvoiceForm>;
+};
+
+export default function AmountForm({ register, control }: AmountsProps) {
+  const { fields, remove, append } = useFieldArray<IInvoiceForm>({
+    control,
+    name: 'amounts',
+  });
+
+  return (
+    <>
+      {fields.map((field, index) => (
+        <>
+          <Grid item container spacing={3}>
```

Comment:
You are missing the `key` prop on array items. (https://kentcdodds.com/blog/understanding-reacts-key-prop)